### PR TITLE
zeroex - refactor Optimism

### PR DESCRIPTION
**Pull request by RantumBits**
# Thank you for contributing to Spellbook!
Please refer to the top of the `readme` in the root of Spellbook to learn how to contribute to Spellbook on DuneSQL.

### CODE REVIEW

File: models/zeroex/optimism/zeroex_optimism_api_fills.sql [hunk 1]
```diff
@@ -96,18 +96,18 @@ v4_limit_fills_no_bridge AS (
             COALESCE(zeroex_tx.affiliate_address, fills.feeRecipient) AS affiliate_address,
             (zeroex_tx.tx_hash IS NOT NULL) AS swap_flag,
             (fills.feeRecipient in 
-                ('0x9b858be6e3047d88820f439b240deac2418a2551','0x86003b044f70dac0abc80ac8957305b6370893ed','0x5bc2419a087666148bfbe1361ae6c06d240c6131')) 
+                (0x9b858be6e3047d88820f439b240deac2418a2551,0x86003b044f70dac0abc80ac8957305b6370893ed,0x5bc2419a087666148bfbe1361ae6c06d240c6131)) 
                 AS matcha_limit_order_flag 
     FROM {{ source('zeroex_optimism', 'ExchangeProxy_evt_LimitOrderFilled') }} fills
 
     INNER JOIN zeroex_tx ON zeroex_tx.tx_hash = fills.evt_tx_hash
```
Comment (Hosuke) [inline models/zeroex/optimism/zeroex_optimism_api_fills.sql:103]: There are only 97 records in zeroex_tx CTE:
https://dune.com/queries/3041602

File: models/zeroex/optimism/zeroex_optimism_api_fills.sql [hunk 2]
```diff
@@ -18,37 +20,35 @@ WITH zeroex_tx AS (
...
                 -- forwarder addresses
-                '0x6958f5e95332d93d21af0d7b9ca85b8212fee0a5',
-                '0x4aa817c6f383c8e8ae77301d18ce48efb16fd2be',
-                '0x4ef40d1bf0983899892946830abf99eca2dbc5ce', 
+                0x6958f5e95332d93d21af0d7b9ca85b8212fee0a5,
+                0x4aa817c6f383c8e8ae77301d18ce48efb16fd2be,
+                0x4ef40d1bf0983899892946830abf99eca2dbc5ce,
                 -- exchange proxy
-                '0xdef1abe32c034e558cdd535791643c58a13acc10'
+                0xdef1c0ded9bec7f1a1670819833240f027b25eff
```
Comment (Hosuke) [inline models/zeroex/optimism/zeroex_optimism_api_fills.sql:40]: If we change this line back to `0xdef1abe32c034e558cdd535791643c58a13acc10`,
the `zeroex_tx` cte will have the tx_hash in the seed file:
https://dune.com/queries/3042243
I am not sure if this line is a typo, but I will change back for now.
```suggestion
                0xdef1abe32c034e558cdd535791643c58a13acc10
```

File: models/zeroex/optimism/zeroex_optimism_api_fills.sql [hunk 3]
```diff
@@ -96,18 +96,18 @@ v4_limit_fills_no_bridge AS (
             COALESCE(zeroex_tx.affiliate_address, fills.feeRecipient) AS affiliate_address,
             (zeroex_tx.tx_hash IS NOT NULL) AS swap_flag,
             (fills.feeRecipient in 
-                ('0x9b858be6e3047d88820f439b240deac2418a2551','0x86003b044f70dac0abc80ac8957305b6370893ed','0x5bc2419a087666148bfbe1361ae6c06d240c6131')) 
+                (0x9b858be6e3047d88820f439b240deac2418a2551,0x86003b044f70dac0abc80ac8957305b6370893ed,0x5bc2419a087666148bfbe1361ae6c06d240c6131))
```
Comment (Hosuke) [inline models/zeroex/optimism/zeroex_optimism_api_fills.sql:99]: I am not sure if the third address is a typo during migration:
```suggestion
                (0x9b858be6e3047d88820f439b240deac2418a2551,
                0x86003b044f70dac0abc80ac8957305b6370893ed,
                0x5bc2419a087666148bfbe1361ae6c06d240c6131)) 
```

**Comment by RantumBits** [comment]
all set here @Hosuke

File: models/zeroex/optimism/zeroex_optimism_api_fills.sql [hunk 4]
```diff
@@ -150,168 +152,61 @@ ERC20BridgeTransfer AS (
...
+            bytearray_to_uint256(bytearray_substring(DATA, 77, 20)) AS taker_token_amount_raw,
+            bytearray_to_uint256(bytearray_substring(DATA, 110, 20)) AS maker_token_amount_raw,
             'ERC20BridgeTransfer'                   AS type,
             zeroex_tx.affiliate_address             AS affiliate_address,
             TRUE                                    AS swap_flag,
             FALSE                                   AS matcha_limit_order_flag
     FROM {{ source('optimism', 'logs') }} logs
     INNER JOIN zeroex_tx ON zeroex_tx.tx_hash = logs.tx_hash
-    WHERE topic1 = '0x349fc08071558d8e3aa92dec9396e4e9f2dfecd6bb9065759d1932e7da43b8a9'
+    WHERE topic0 = 0x349fc08071558d8e3aa92dec9396e4e9f2dfecd6bb9065759d1932e7da43b8a9
```
Comment (Hosuke) [inline models/zeroex/optimism/zeroex_optimism_api_fills.sql:167]: `topic1` in spark sql becomes `topic0` in dune sql.
https://dune.com/docs/migrations/SparkSQL/#migrating-from-sparksql

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @RantumBits 

[Sample zeroex_optimism_api_fills]()
[Sample zeroex_optimism_api_fills_deduped]()

**Review by jeff-dude — APPROVED** [review]
lgtm!

thank you @RantumBits for initiating & @Hosuke for finalizing 🚀

**Pull request merged**